src/Native/System.Security.Cryptography.Native/pal_rsa.cpp

```diff
@@ -12,3 +31,122 @@ extern "C" RSA* DecodeRsaPublicKey(const unsigned char* buf, int32_t len)
 
     return d2i_RSAPublicKey(nullptr, &buf, len);
 }
+
+extern "C" int32_t RsaPublicEncrypt(int32_t flen, const uint8_t* from, uint8_t* to, RSA* rsa, int32_t useOaepPadding)
+{
+    int padding = useOaepPadding ? RSA_PKCS1_OAEP_PADDING : RSA_PKCS1_PADDING;
+    return RSA_public_encrypt(flen, from, to, rsa, padding);
+}
+
+extern "C" int32_t RsaPrivateDecrypt(int32_t flen, const uint8_t* from, uint8_t* to, RSA* rsa, int32_t useOaepPadding)
+{
+    int padding = useOaepPadding ? RSA_PKCS1_OAEP_PADDING : RSA_PKCS1_PADDING;
+    return RSA_private_decrypt(flen, from, to, rsa, padding);
+}
+
+extern "C" int32_t RsaSize(RSA* rsa)
+{
+    return RSA_size(rsa);
+}
+
+extern "C" int32_t RsaGenerateKeyEx(RSA* rsa, int32_t bits, BIGNUM* e)
+{
+    return RSA_generate_key_ex(rsa, bits, e, nullptr);
+}
+
+extern "C" int32_t RsaSign(int32_t type, const uint8_t* m, int32_t mlen, uint8_t* sigret, int32_t* siglen, RSA* rsa)
+{
+    if (!siglen)
+    {
+        assert(false);
+        return 0;
```

Comment:
rsa isn't an 'out' argument.  It is just an 'in' pointer.  'siglen' is the only out argument.

if (!siglen), then siglen can't be initialized, right?

and if siglen is a valid pointer, then it will be set below unconditionally.